postinst.d/50-dracut.install

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+ver=${1}
+img=${2}
+initrd="${INSTALLKERNEL_STAGING_AREA:-$(dirname "${img}")}/initrd"
+uki="${INSTALLKERNEL_STAGING_AREA:-$(dirname "${img}")}/uki.efi"
+tmp="${INSTALLKERNEL_STAGING_AREA:-$(dirname "${img}")}/dracut-tmp"
+
+# familiar helpers, we intentionally don't use Gentoo functions.sh
+die() {
+	echo -e " ${NOCOLOR-\e[1;31m*\e[0m }${*}" >&2
+	exit 1
+}
+
+einfo() {
+	echo -e " ${NOCOLOR-\e[1;32m*\e[0m }${*}"
+}
+
+ewarn() {
+	echo -e " ${NOCOLOR-\e[1;33m*\e[0m }${*}"
+}
+
+main() {
+	# re-define for subst to work
+	[[ -n ${NOCOLOR+yes} ]] && NOCOLOR=
+
+	if [[ -n "${INSTALLKERNEL_INITRD_GENERATOR}" &&
+		${INSTALLKERNEL_INITRD_GENERATOR} != dracut ]]; then
+			# We are not selected as the initrd generator
+			einfo "initrd_generator=${INSTALLKERNEL_INITRD_GENERATOR}, skipping dracut"
+			exit 0
+	fi
+
+	# do nothing if somehow dracut is not installed
+	[[ -x $(command -v dracut) ]] || { ewarn "dracut is not installed, not generating an initramfs" && exit 0; }
+
+	[[ ${EUID} -eq 0 ]] || die "Please run this script as root"
+
+	initramfs_gen_args=(
```

Comment:
This script could be a `/bin/sh` script in case you would skip this `initramfs_gen_args` and move it to the dracut call.

Reply:
It could be, but the systemd equivalent of this script is also a `/bin/bash` script, and I'd like to keep the symmetry since this makes it easier to make identical changes to both files.

Reply:
If we want to POSIXify these scripts than I propose we do this simultaneously with identical changes for the systemd version in a separate PR (also adding then `set -eu` like you suggested).